router/src/validation.rs

```diff
@@ -536,6 +545,140 @@ fn format_to_mimetype(format: ImageFormat) -> String {
     .to_string()
 }
 
+#[cfg(not(feature = "video"))]
+pub fn fetch_video(
+    _input: &str,
+    _target_width: u32,
+    _target_height: u32,
+) -> Result<ProcessedVideo, ValidationError> {
+    Err(ValidationError::VideoNotSupported)
+}
+
+#[cfg(feature = "video")]
+pub fn fetch_video(
```

Comment:
This overall seems super convoluted.

- Why do we need a file for something that should stay in RAM ?
- We are re-encoding the video here, this is fine if we are agressively trimming the output video (like removing extra frames). We have to assume ppl are going to send 4k video of 10hour long content.
- We are binding to `ffmpeg` which seems to add a lot of dependency complexity. Why not just depend on `ffmpeg` binary, and be done with it ? Seems much simpler, and we're not looking at all at the contents it seems.

For the tempfile, it *might* be the simplest. But I feel we need to discuss options before doing it that way.